src/ray/core_worker/actor_manager.cc

```diff
@@ -299,5 +286,26 @@ ActorID ActorManager::GetCachedNamedActorID(const std::string &actor_name) {
   return ActorID::Nil();
 }
 
+void ActorManager::SubscribeActorState(const ActorID &actor_id) {
+  auto actor_handle = GetActorHandle(actor_id);
+  RAY_CHECK(actor_handle != nullptr);
+  if (actor_handle->IsActorStateSubscribed()) {
+    return;
+  }
+
+  absl::MutexLock lock(&subscribe_mutex_);
+
```

Comment:
Should check `actor_handle->IsActorStateSubscribed()` again right after the lock is acquired.